### PR TITLE
Fix/2.0 tests

### PR DESCRIPTION
**Pull request by soyuka**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | na
| License       | MIT
| Doc PR        | na

### CODE REVIEW

File: src/Metadata/Extractor/YamlExtractor.php [hunk 1]
```diff
@@ -32,7 +32,7 @@
     protected function extractPath(string $path)
     {
         try {
-            $resourcesYaml = Yaml::parse(file_get_contents($path));
+            $resourcesYaml = Yaml::parse(file_get_contents($path), Yaml::PARSE_KEYS_AS_STRINGS);
```
Comment (soyuka) [inline src/Metadata/Extractor/YamlExtractor.php]: Not sure about this one, @meyerbaptiste is this correct? Thanks!

Comment (meyerbaptiste) [inline src/Metadata/Extractor/YamlExtractor.php]: Yes it's correct, but this constant was introduced in `symfony/yaml` 3.3. So tests will fail with the `--prefer-lowest` argument!

Comment (soyuka) [inline src/Metadata/Extractor/YamlExtractor.php]: Hmm, what solution do we have if we want tests to pass on 2.0 then?

Comment (soyuka) [inline src/Metadata/Extractor/YamlExtractor.php]: I could mark tests as legacy but doesn't feel clean enough.

Comment (dunglas) [inline src/Metadata/Extractor/YamlExtractor.php]: IIRC, using this constant is deprecated, the user should use quotes instead.

Comment (soyuka) [inline src/Metadata/Extractor/YamlExtractor.php]: hmm okay maybe I need to fix yaml files instead then.

File: tests/Fixtures/FileConfigurations/resources.yml [hunk 2]
```diff
@@ -32,7 +32,7 @@ resources:
                 attributes:
                     'foo': ['Foo']
                     'bar':
-                        0: ['Bar']
+                        '0': ['Bar']
```
Comment (soyuka) [inline tests/Fixtures/FileConfigurations/resources.yml]: thanks @dunglas :smile:

File: src/Metadata/Extractor/YamlExtractor.php (hunk 1 above)
Comment (meyerbaptiste) [inline src/Metadata/Extractor/YamlExtractor.php]: > IIRC, using this constant is deprecated

Are you sure? I can't find any information on this subject and I don't see anything in the code of the component.

Comment (soyuka) [inline src/Metadata/Extractor/YamlExtractor.php]: Anyway, fixing the actual issue in the yaml seems to be the way to go.

Comment (dunglas) [inline src/Metadata/Extractor/YamlExtractor.php]: @meyerbaptiste https://github.com/symfony/symfony/pull/22948

**Pull request merged**